include/circt/Dialect/MSFT/MSFTOps.td

```diff
@@ -38,3 +38,83 @@ def InstanceOp : MSFTOp<"instance", [
       `:` functional-type($inputs, results)
   }];
 }
+
+def OneOrNoBlocksRegion : Region<
+  CPred<"::llvm::hasNItemsOrLess($_self, 1)">,
+  "region with at most 1 block">;
+
+def MSFTModuleOp : MSFTOp<"module",
```

Comment:
I'm not totally familiar with this dialect, but if an InstanceOp can be either an instance of a MSFT Module or a HW Module, would it make sense to use [ParentOneOf](https://mlir.llvm.org/docs/Traits/#hasparent)?

Reply:
You mean on the `InstanceOp`? Yeah, I think that makes sense, but not strictly related to this PR.